src/device/wg_controller.rs

```diff
@@ -551,7 +579,12 @@ async fn build_requested_peers_list<
                 ip_addresses.push(VPN_INTERNAL_IPV6.into());
             }
             ip_addresses.push(VPN_EXTERNAL_IPV4.into());
-            let persistent_keepalive_interval = requested_state.keepalive_periods.vpn;
+            let (persistent_keepalive_interval, batching_keepalive_interval) =
+                if features.batching.is_some() {
+                    (None, requested_state.keepalive_periods.vpn)
+                } else {
+                    (requested_state.keepalive_periods.vpn, None)
+                };
```

Comment:
I am thinking about this improvement and cannot understand if it's safely implemented. Given that tests are passing it probably is, however the switching of the values is what gets me a bit paranoid, also it happens not once. One way to secure against it would be pass values straight and nearby UAPI check if we have batching then don't use the regular persistent-ka value. This of course isn't also perfect as then the logic for batching spills a bit into another area but might be a less complex solution than the current one. What are your thoughts on this one?